### PR TITLE
add CURLSSLBACKEND_ALL enum

### PR DESCRIPTION
**Pull request by TedLyngmo**
[ssl]: add CURLSSLBACKEND_ALL enum for cleaner code and examples

This is to make code that now does `(curl_sslbackend)-1` clearer
and easier to read:
```
/* list the available ones */
const curl_ssl_backend **list;
curl_global_sslset(CURLSSLBACKEND_ALL, NULL, &list);
```

### CODE REVIEW

File: include/curl/curl.h [hunk 1]
```diff
@@ -170,7 +170,8 @@ typedef enum {
   CURLSSLBACKEND_MBEDTLS = 11,
   CURLSSLBACKEND_MESALINK               CURL_DEPRECATED(7.82.0, "") = 12,
   CURLSSLBACKEND_BEARSSL = 13,
-  CURLSSLBACKEND_RUSTLS = 14
+  CURLSSLBACKEND_RUSTLS = 14,
+  CURLSSLBACKEND_ALL = ~0
```
Comment (bagder) [inline include/curl/curl.h:174]: I think it should be set to `-1` and be made first in the list.

**Review by bagder — COMMENTED** [review]
Also, you did not update `symbols-in-versions` so test 1119 fails.

File: include/curl/curl.h (hunk 1 above)
Comment (TedLyngmo) [reply include/curl/curl.h:174]: Ok, but why prefer `-1`? We want all bits set, which `~0` guarantees.

**Comment by TedLyngmo** [comment]
> symbols-in-versions

Thanks! What will the next version be? 8.3.1?

File: include/curl/curl.h (hunk 1 above)
Comment (bagder) [reply include/curl/curl.h:174]: It is not used like that. It is just supposed to be a number that does not match one of the existing (or future) TLS backends. See:

https://github.com/curl/curl/blob/3d53f211e5c907bb6e8aaddf05a3acdcae247e5d/lib/vtls/vtls.c#L1369-L1396

**Comment by bagder** [comment]
> Thanks! What will the next version be? 8.3.1?

We are fairly sure it will become 8.4.0 so I think you can use that.

**Comment by jay** [comment]
IMO `curl_global_sslset(CURLSSLBACKEND_ALL, ...` is confusing because it implies setting all backends. To me, it makes more sense as -1.

**Comment by bagder** [comment]
> CURLSSLBACKEND_ALL

Yeah, maybe *_NONE instead?

**Comment by TedLyngmo** [comment]
> > CURLSSLBACKEND_ALL
> 
> Yeah, maybe *_NONE instead?

`CURLSSLBACKEND_NONE` is already defined to `0`. I used `_ALL` because it seemed to fit into the documentation:

```
/* list the available ones */
const curl_ssl_backend **list;
curl_global_sslset((curl_sslbackend)-1, NULL, &list);
````

**Comment by bagder** [comment]
> `CURLSSLBACKEND_NONE` is already defined to `0`. I used `_ALL` because it seemed to fit into the documentation:

Oh right. But maybe we should then rather change the documentation to use *NONE instead of -1? As far as I can see, that gives the same result...

**Comment by TedLyngmo** [comment]
> > `CURLSSLBACKEND_NONE` is already defined to `0`. I used `_ALL` because it seemed to fit into the documentation:
> 
> Oh right. But maybe we should then rather change the documentation to use *NONE instead of -1? As far as I can see, that gives the same result...

Ah, well, that would certainly be even easier if it indeed does the same thing :-)

**Comment by TedLyngmo** [comment]
Since #11909 has been approved, this PR is meaningless. Closing.

**Pull request closed without merging**